client/components/reviews-rating-stars/reviews-ratings-stars.tsx

```diff
@@ -0,0 +1,145 @@
+import { useTranslate } from 'i18n-calypso';
+import { useState, useEffect } from 'react';
+import Rating from 'calypso/components/rating';
+import Star, { type StarProps } from './star';
+import './reviews-ratings-stars.scss';
+
+export const MAX_RATING = 5;
+
+type ReviewsRatingsStarsProps = {
+	rating: number;
+	averageRating?: number;
+	numberOfReviews?: number;
+	size?: 'medium' | 'medium-small' | 'medium-large';
+	showSelectedRating?: boolean;
+	children?: React.ReactNode;
+	onMouseEnter?: () => void;
+	onMouseLeave?: () => void;
+	onSelectRating?: ( index: number ) => void;
+	simpleView?: boolean;
+};
+
+const ReviewsRatingsStars = ( props: ReviewsRatingsStarsProps ) => {
+	const {
+		rating,
+		averageRating,
+		numberOfReviews,
+		size = 'medium',
+		showSelectedRating = false,
+		children,
+		onMouseEnter,
+		onMouseLeave,
+		onSelectRating,
+		simpleView,
+	} = props;
+
+	const translate = useTranslate();
+
+	const [ ratingState, setRatingState ] = useState< number >( rating );
+	const [ hoverRating, setHoverRating ] = useState( rating );
+
+	const onStarMouseEnter = ( index: number ) => setHoverRating( index );
+	const onStarMouseLeave = () => setHoverRating( ratingState );
+	const onSaveRating = ( _: unknown, index: number ) => {
+		setRatingState( index );
+		if ( onSelectRating ) {
+			onSelectRating( index );
+		}
+	};
+
+	useEffect( () => {
+		setRatingState( rating );
+		setHoverRating( rating );
+	}, [ rating ] );
+
+	const classNames = [ 'reviews-ratings-stars__star-bar' ];
+
+	switch ( size ) {
+		case 'medium-small':
+			classNames.push( 'reviews-ratings-stars__star-bar--medium-small' );
+			break;
+		case 'medium-large':
+			classNames.push( 'reviews-ratings-stars__star-bar--medium-large' );
+			break;
+	}
+
+	// Show numberOfReviews if the prop was passed
+	let numberOfReviewsText;
+	if ( numberOfReviews ) {
+		numberOfReviewsText = translate( '%(numberOfReviews) review', '%(numberOfReviews) reviews', {
```

Comment:
Is this translation working? I expect it to need something like `ranslate( '%(numberOfReviews)d review',...` with the `d` or `s` after the parenthesis?

Reply:
Good catch, this is not used at the moment by the consumer but it has the bug you mention that I will fix

Reply:
Fixed here d0818bd3256e92faa2659619e1a95a469858157c